### PR TITLE
ER modeling 

### LINKED ISSUE #19: ERD modeling
# Description 

Design ERD relational database layout using `PostGradSQL`

## Acceptable Criteria 

- Memaid Documentation 
- Explain the entity and their relationship (primary key, foreign key, one-to-many, many-to-many, many-to-one, etc..)
- Make sure that it links up with the UML Design we have. 

## Related Story 

This ticket have to closely tights with #18 

## Technical Info

[Mermaid diagram](https://mermaid-js.github.io/mermaid/#/entityRelationshipDiagram)
[Read](https://www.datanamic.com/support/lt-dez005-introduction-db-modeling.html)

### PR DESCRIPTION
**Pull request by BAbate**
# Description

This is basic ER relationship diagram that is based on UML diagram. It's dependent on MermaidJS

Fixes #19 

## Type of change

Please delete options that are not relevant.

- new features 
- this change itself is documentation that developers refer to construct the database 

# How Has This Been Tested?

It has been tested locally and on live editor. A link is included to help the developer replicate the same test 

**Test Configuration**:
N/A

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [N/A] My changes generate no new warnings
- [N/A] I have added tests that prove my fix is effective or that my feature works
- [N/A] New and existing unit tests pass locally with my changes
- [N/A] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by kirubeltadesse** [comment]
Nice job BK. Could be removed this file
[Doc/.UML_diagram.md.swp](https://github.com/Swarmies/sap-web/pull/32/files#diff-67a0786a8ce700d94a6c17142b8e15e1ea67e10bf99cd63701b2cc3c41f23ae7)

I believe your branch is not up to date. Please do a rebase to fix that!

File: Doc/ER_diagram.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# This diagram represents the ER relationship diagram among the entities below
```
Comment (Getnet12) [inline Doc/ER_diagram.md]: Add a title and make the rest of the information a paragraph: for better formatting.

File: Doc/ER_diagram.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            string userName
+        }
+        STUDENT {
+            int student_id PK
+            string user_id FK
+        }
+        MENTOR {
+            int mentor_id PK
+            string user_id FK
+            string course
```
Comment (Getnet12) [inline Doc/ER_diagram.md]: change it courses instead of course.

File: Doc/ER_diagram.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        ADMIN {
+             int admin_id PK
+             string user_id FK
+        }
+        SIGNIN {
```
Comment (kirubeltadesse) [inline Doc/ER_diagram.md:39]: Let's remove this table because it is redundant.

File: Doc/ER_diagram.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  erDiagram
+          USER ||--|| STUDENT : is
+          USER ||--|| MENTOR : is
+          USER ||--|| ADMIN : is
+          USER ||--|| SIGNIN : has
+          STUDENT ||--o{ EVENTS : posts
+          STUDENT ||--|| CALENDAR : has
+          MENTOR ||--|| CALENDAR : has
+          MENTOR ||--|{ WEEKDAYS : tutors
+          ADMIN ||--o{ EVENTS : posts
```
Comment (kirubeltadesse) [inline Doc/ER_diagram.md]: add a `string password` field here

**Review by kirubeltadesse — COMMENTED** [review]
Here is an additional comment

**Review by Getnet12 — COMMENTED** [review]
It's a suggestion Kira come up with while we were talking.

File: Doc/ER_diagram.md (hunk 1 above)
Comment (BAbate) [reply Doc/ER_diagram.md]: Okay good

File: Doc/ER_diagram.md (hunk 2 above)
Comment (BAbate) [reply Doc/ER_diagram.md]: Good.

File: Doc/ER_diagram.md (hunk 4 above)
Comment (BAbate) [reply Doc/ER_diagram.md]: The first section represents Entity relationships and each constitutes its own table. Do you mean to have a Password entity? or if you are referring to string password, let me know the table so I can add it@kirubeltadesse

File: Doc/ER_diagram.md (hunk 3 above)
Comment (BAbate) [reply Doc/ER_diagram.md:39]: okay good.

File: Doc/ER_diagram.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          CALENDAR ||--|{ WEEKDAYS: has
+        USER {
+            int user_id PK
+            string fullName
+            string Jnumber
+            string email
+            string role
+            string userName
```
Comment (kirubeltadesse) [inline Doc/ER_diagram.md:27]: Sorry, I mean add the password here. Next to the `userName`

File: Doc/ER_diagram.md (hunk 4 above)
Comment (kirubeltadesse) [reply Doc/ER_diagram.md]: I understand. clicked at the wrong place by mistake :)

File: Doc/ER_diagram.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            string mentor_id FK
+            string student_id FK
+            dateFormat startDate
+            dateFormat endDate
+        }
+        CALENDAR {
+            int calendar_id PK
+            int weekday_id FK
+        }
+        WEEKDAYS {
```
Comment (dinaolmelak) [inline Doc/ER_diagram.md]: Screen Shot 2022-06-06 at 12 34 25 PM

Should we add an entry for 'week' here?

File: Doc/ER_diagram.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            string password
+        }
+        EVENTS {
+            int events_id PK
+            string mentor_id FK
+            string student_id FK
+            dateFormat startDate
+            dateFormat endDate
+        }
+        CALENDAR {
```
Comment (dinaolmelak) [inline Doc/ER_diagram.md:52]: Doesn't the admin need access to the calendar db?

**Review by dinaolmelak — CHANGES_REQUESTED** [review]

File: Doc/ER_diagram.md (hunk 7 above)
Comment (kirubeltadesse) [reply Doc/ER_diagram.md:52]: That will be definitely higher on the MVC level, not on the DB level!

File: Doc/ER_diagram.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+```mermaid
+  erDiagram
+          USER ||--|| STUDENT : is 
+          USER ||--|| MENTOR : is
+          USER ||--|| SIGNIN : has
+          STUDENT ||--o{ EVENTS : posts
+          STUDENT ||--|| CALENDAR : has
+          MENTOR ||--|| CALENDAR : has
+        USER {
+            int user_id PK
```
Comment (dinaolmelak) [inline Doc/ER_diagram.md:22]: The datatype of user_id seems to be inconsistent. It is an int datatype here but on STUDENT, MENTOR and SIGNIN it seems to be of string data type

File: Doc/ER_diagram.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            string fullName
+            string Jnumber
```
Comment (dinaolmelak) [inline Doc/ER_diagram.md:24]: can we set this to camel case naming convention? (i.e string jNumber)

File: Doc/ER_diagram.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+            string role
+            string userName
+            string password
+        }
+        STUDENT {
+            int student_id PK
+            string user_id FK
+        }
+        MENTOR {
+            int mentor_id PK
```
Comment (dinaolmelak) [inline Doc/ER_diagram.md:35]: The data type of `mentor_id` is int here but under EVENTS it is a string.

File: Doc/ER_diagram.md (hunk 9 above)
Comment (dinaolmelak) [inline Doc/ER_diagram.md:24]: Can we also set the data type of jNumber to `int`?

File: Doc/ER_diagram.md (hunk 8 above)
Comment (kirubeltadesse) [reply Doc/ER_diagram.md:22]: @dinaolmelak this is closed PR?

Comment (dinaolmelak) [reply Doc/ER_diagram.md:22]: Gotcha, I have added these changes to my pull request #44

**Pull request closed without merging**